worldedit-fabric/src/main/java/com/sk89q/worldedit/fabric/FabricWorldEdit.java

```diff
@@ -171,6 +181,43 @@ public void onInitialize() {
 
         WECUIPacketHandler.init();
 
+        // Register the CUI plugin channel
+        ServerTickEvents.END_SERVER_TICK.register(server -> {
+            long currentTime = System.currentTimeMillis();
+
+            if (currentTime - lastTickUpdate < 200) { // Throttle updates to 200ms
+                return;
+            }
+            lastTickUpdate = currentTime;
+
+            for (ServerPlayer player : server.getPlayerList().getPlayers()) {
+                FabricPlayer wePlayer = FabricAdapter.adaptPlayer(player);
+                LocalSession session = WorldEdit.getInstance().getSessionManager().get(FabricAdapter.adaptPlayer(player));
+
+                ItemStack heldItem = player.getMainHandItem();
+                Tool tool = session.getTool(FabricAdapter.adapt(heldItem.getItem()));
+
+                if (tool instanceof BrushTool) {
+                    Vec3 currentLookDirection = getPlayerLook(player);
+
+                    UUID playerUUID = player.getUUID();
+                    Vec3 lastLookDirection = playerLookDirections.get(playerUUID);
+
+                    if (currentLookDirection.equals(lastLookDirection)) {
+                        continue;
+                    }
```

Comment:
In case the player moves but still looks in the same direction, will it re-render the preview?

Reply:
Esta certo.